scripts/zones/Temenos/mobs/Telchines_Monk.lua

```diff
@@ -2,20 +2,45 @@
 -- Area: Temenos N T
 --  Mob: Telchines Monk
 -----------------------------------
-require("scripts/globals/limbus");
------------------------------------
-
-function onMobEngaged(mob,target)
+require("scripts/globals/limbus")
+require("scripts/globals/pathfind")
+mixins = {require("scripts/mixins/job_special")}
+local ID = require("scripts/zones/Temenos/IDs")
+local flags = tpz.path.flag.NONE
+local path =
+{
+    [4] = 
+    {
+        {30.000, 80.000, 420.500},
+        {10.000, 80.000, 420.500}
+    },
+    [5] = 
+    {
+        {70.000, 80.000, 420.500},
+        {50.000, 80.000, 420.500}
+    },
+}
 
-end;
+function onMobRoam(mob)
+    local offset = mob:getID() - ID.mob.TEMENOS_N_MOB[3]
+    local pause = mob:getLocalVar("pause")
+    if pause < os.time() then
+        local point = (mob:getLocalVar("point") % 2)+1
+        mob:setLocalVar("point", point)
+        mob:pathTo(path[offset][point][1], path[offset][point][2], path[offset][point][3], flags)
+        mob:setLocalVar("pause", os.time()+10)
+    end
+end
 
 function onMobDeath(mob, player, isKiller)
- if (IsMobDead(16928788)==true and IsMobDead(16928789)==true  and IsMobDead(16928792)==true   and IsMobDead(16928793)==true ) then
-       GetNPCByID(16928768+26):setPos(19,80,430);
-    GetNPCByID(16928768+26):setStatus(tpz.status.NORMAL);
-    GetNPCByID(16928768+160):setPos(16,80,430);
-    GetNPCByID(16928768+160):setStatus(tpz.status.NORMAL);
-    GetNPCByID(16928768+211):setPos(22,80,430);
-    GetNPCByID(16928768+211):setStatus(tpz.status.NORMAL);
- end
-end;
+    if isKiller then
+        local mobID = mob:getID()
+        local battlefield = player:getBattlefield()
+        local random = battlefield:getLocalVar("randomF3")
+
+        if mobID - ID.mob.TEMENOS_N_MOB[3] == random - 1 then
```

Comment:
Is this a `random == 5` case?

Reply:
I'm not sure what I did here, but it looks wrong. `random` can be 1-6, but doors won't open if it's 1 or 2. Will fix.